android/app/src/androidTest/java/com/trustwallet/core/app/blockchains/filecoin/TestFilecoin.kt

```diff
@@ -20,13 +22,30 @@ class TestFilecoin {
     }
 
     @Test
-    fun testAddress() {
+    fun testCreateAddress() {
         val privateKey = PrivateKey("1d969865e189957b9824bd34f26d5cbf357fda1a6d844cbf0c9ab1ed93fa7dbe".toHexByteArray())
         val publicKey = privateKey.getPublicKeySecp256k1(false)
         val address = AnyAddress(publicKey, CoinType.FILECOIN)
         assertEquals("f1z4a36sc7mfbv4z3qwutblp2flycdui3baffytbq", address.description())
     }
 
+    @Test
+    fun testCreateDelegatedAddress() {
+        val privateKey = PrivateKey("825d2bb32965764a98338139412c7591ed54c951dd65504cd8ddaeaa0fea7b2a".toHexByteArray())
+        val publicKey = privateKey.getPublicKeySecp256k1(false)
+        val address = AnyAddress(publicKey, FilecoinAddressType.DELEGATED)
+        assertEquals("f410fvak24cyg3saddajborn6idt7rrtfj2ptauk5pbq", address.description())
+    }
+
+    @Test
+    fun testAddressConverter() {
+        val ethereumAddress = FilecoinAddressConverter.convertToEthereum("f410frw6wy7w6sbsguyn3yzeygg34fgf72n5ao5sxyky")
+        assertEquals(ethereumAddress, "0x8dbD6c7Ede90646a61Bbc649831b7c298BFd37A0")
```

Comment:
I would add an assertion that the ethereumAddress is valid.

AnyAddress.isvalid(address, coin);